### PR TITLE
Fix issue when check sample solution name

### PR DESCRIPTION
**Pull request by SLdragon**
https://github.com/Microsoft/vscode-azure-iot-edge/issues/379

### CODE REVIEW

File: src/edge/edgeManager.ts [hunk 1]
```diff
@@ -609,6 +609,22 @@ export class EdgeManager {
         return undefined;
     }
 
+    private async validateSolutionName(name: string, parentPath?: string): Promise<string | undefined> {
+        if (!name) {
+            return "The name could not be empty";
+        }
+        if (name.match(/[/\\:*?\"<>|]/)) {
+            return "Solution name cannot contain special characters";
```
Comment (LazarusX) [inline src/edge/edgeManager.ts]: "Special characters" here is not clear enough. I prefer something like "Only *** are allowed".

Comment (SLdragon) [reply src/edge/edgeManager.ts]: User can use Chinese or other utf8 characters for folder names, so I think "Only..." is not easy to describe the limitation

File: src/edge/edgeManager.ts [hunk 2]
```diff
@@ -609,6 +609,22 @@ export class EdgeManager {
         return undefined;
     }
 
+    private async validateSolutionName(name: string, parentPath?: string): Promise<string | undefined> {
+        if (!name) {
```
Comment (adashen) [inline src/edge/edgeManager.ts]: name could be spaces? So we need to trim the spaces

File: src/edge/edgeManager.ts (hunk 1 above)
Comment (adashen) [reply src/edge/edgeManager.ts]: How about change the message to Solution name can't contain characters: \/:*?<>|

File: src/edge/edgeManager.ts (hunk 2 above)
Comment (SLdragon) [reply src/edge/edgeManager.ts]: Oh, yes, I will trim it.

File: src/edge/edgeManager.ts (hunk 1 above)
Comment (SLdragon) [reply src/edge/edgeManager.ts]: Thanks, fixed

**Comment by SLdragon** [comment]
Also verified
https://github.com/Microsoft/vscode-azure-iot-edge/issues/74
https://github.com/Microsoft/vscode-azure-iot-edge/issues/75
https://github.com/Microsoft/vscode-azure-iot-edge/issues/76

**Pull request merged**